src/thinblock.cpp

```diff
@@ -312,34 +311,34 @@ bool CXThinBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom, string strComm
 
         inv.hash = pIndex->GetBlockHash();
 
-        // Request thin block if it isn't extending the best chain
+        // Request full block if it isn't extending the best chain
         if (pIndex->nChainWork <= chainActive.Tip()->nChainWork)
         {
             vector<CInv> vGetData;
-            vGetData.push_back(CInv(MSG_THINBLOCK, inv.hash));
-
+            vGetData.push_back(inv);
```

Comment:
i believe this is incorrect, you've formatted previously a inv with no hash...you have to give it the block hash.  i would put here the whole string so there is no confusion

`+ vGetData.push_back(CInv(MSG_BLOCK, inv.hash));`